pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -86,13 +87,13 @@ var _ CIDRAllocator = (*cloudCIDRAllocator)(nil)
 // NewCloudCIDRAllocator creates a new cloud CIDR allocator.
 func NewCloudCIDRAllocator(client clientset.Interface, cloud cloudprovider.Interface, nodeInformer informers.NodeInformer) (CIDRAllocator, error) {
 	if client == nil {
-		klog.Fatalf("kubeClient is nil when starting NodeController")
+		return nil, errors.New("kubeClient is nil when starting cloud CIDR allocator")
```

Comment:
Isn't this a behavior change? I'm not sure we want to hide this in what is being described as a logging change. I would hold on off on these changes and put it in a separate PR.

Reply:
That makes sense. This PR then depends on that other one.

Reply:
yeah, I have updated all.